### ISSUE TITLE
Missing many items.

### ISSUE THREAD

**Issue by manuramesh96** [Wormgit/LabelaHerdinMinutes#1]
I guess the full code was not uploaded.
I found many issues in the training scripts. But I was able to fix them or find work arounds. (For fully supervised training).

Validation:
- Does not work.

For testing, 
- Missing KNNAccuracy function in test_utils.py.
- test_mine_EmbMulti.py does not run because there are no [KNN ground truth labels.](https://github.com/Wormgit/LabelaHerdinMinutes/blob/8025723cbacfcf280c55535d425a3c8e6867d7e0/src/MetricLearning/test_mine_EmbMulti.py#L66) 

Can these be fixed? Is there a way to train and evaluate on custom datasets?
